src/components/lib/crt/crt.c

```diff
@@ -656,7 +1018,13 @@ crt_thd_create_in(struct crt_thd *t, struct crt_comp *c, thdclosure_index_t clos
 
 		crt_refcnt_take(&c->refcnt);
 		assert(target_ci->comp_cap);
-		thdcap = target_aep->thd = cos_initthd_alloc(ci, target_ci->comp_cap);
+		/* FIXME: assumes comp cap shared would be originally initialized to 0 */
+		if(target_ci->comp_cap_shared != 0) {
```

Comment:
Hard for me to see the combinatorics behind this. Hard for me to think through all of the cases this will be true. If it is simple, ignore me ;-)